src/client/meshgen/extrusion.cpp

```diff
@@ -0,0 +1,55 @@
+#include "extrusion.h"
+#include <map>
+#include <SMeshBuffer.h>
+#include "extruder.h"
+
+static std::map<video::ITexture *, ExtrudedMesh> cache;
```

Comment:
Is it a good idea to store this as a global variable? In my opinion creating a cache class and adding it to Client as a member (like `TextureSource`, `ShaderSource`) is a better idea (yes, I know that it was global before too).

Reply:
The only point for that may be readability, as it won’t work any better than the current code. But having to pass `client` here and there all the time is too annoying IMO. As well as using interfaces that have only one implementation, and are very unlikely to get another one.